core/core.go

```diff
@@ -250,6 +257,42 @@ func initPrivateKey(path string) (p2pcrypto.PrivKey, error) {
 	return nil, err
 }
 
+func initNetworkId(networkID int, db *meshdb.MeshDB) error {
+	// see if there is already a stored networkID
+	snapshot, err := db.Metadata.GetSnapshot()
```

Comment:
You don't need to use a snapshot here. The [doc comment for `Snapshot`](https://godoc.org/github.com/0xProject/0x-mesh/db#Snapshot) explains:

> Snapshot is a frozen, read-only snapshot of a DB state at a particular point in time.

Snapshots are only really needed when other goroutines can change the state of the database and you want a single consistent view which ignores those changes. For example, we use snapshots for the `GetOrders` JSON-RPC API endpoint, because we don't want the orders to change while you're in the middle of paginating through them.

In this case, no other goroutine is going to change the metadata, so we don't need to call `GetSnapshot`.

Reply:
Makes sense, thanks for the tip. I was using some other methods as a guide, and assumed a snapshot should be used for all reads. will make this fix!